cmake/compiler/gcc/compiler_flags.cmake

```diff
@@ -180,6 +180,9 @@ check_set_compiler_property(PROPERTY freestanding -ffreestanding)
 # Flag to enable debugging
 set_compiler_property(PROPERTY debug -g)
 
+# Flags to save temporary object files
+set_compiler_property(PROPERTY save_temps -save-temps=obj)
```

Comment:
clang doc only mentions a pure `-save-temps`, Ref: https://clang.llvm.org/docs/CommandGuide/clang.html#cmdoption-save-temps

although in practice clang seems to also accept `-save-temps=obj`, then I believe safest is to adjust the property here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/compiler/clang/compiler_flags.cmake

And derived armclang:
https://developer.arm.com/documentation/101754/0620/armclang-Reference/armclang-Command-line-Options/-save-temps

Reply:
Cool, added the clang bit.